### PR TITLE
Adding functionality to GroceryListService and connecting with Household

### PR DESCRIPTION
**Pull request by rgoonie**
Adding functionality to GroceryListService and connecting the service with the HouseholdService. Test written for both GroceryList and Household updates

### CODE REVIEW

File: src/services/grocerylist/GroceryListService.js [hunk 1]
```diff
@@ -1,67 +1,273 @@
...
-     * @param {string} ownerId 
+     * Creates a new grocery list and returns the grocery list object with an id
+     * @param {string} name name of the grocery list
+     * @param {string} ownerId the household/person the container belongs to
+     * @returns {GroceryList} the groceryList object that was created
+     * @throws error if grocery list was not able to be made in DB
      */
-    createGroceryList(name, ownerId){
-        throw new Error('Not Implemented')
+    async createGroceryList(name, ownerId = null){
```
Comment (ynigoreyes) [inline src/services/grocerylist/GroceryListService.js:16]: So why would we ever have an ownerId of null? Could explain your thought process on this?

File: src/services/grocerylist/GroceryListService.js [hunk 2]
```diff
@@ -1,67 +1,273 @@
...
     }
 
     /**
      * Finds and returns a list of food items who name
      *      matches the search
-     * @param {String} itemName
+     * @param {string} id the id of the grocery list being searched
+     * @param {string} itemName the name of the item being search for
      * @returns a list of food items 
      */
```
Comment (ynigoreyes) [inline src/services/grocerylist/GroceryListService.js:251]: I always shy away from using regex, but if it is thoroughly tested, I don’t see why not. Please make sure to add a comment as to what this regex is looking for. No one likes reading regex test <3

**Review by ynigoreyes — COMMENTED** [review]
Good work. Thank you for writing good tests for the search feature. That wizardry they call regex always scares me. However I would consider at least looking into just straight up using `string.includes(String matchString)`

File: src/services/grocerylist/GroceryListService.js (hunk 1 above)
Comment (rgoonie) [reply src/services/grocerylist/GroceryListService.js:16]: This is preparation for doing personal/private grocery list. If the grocery list is public to the household then there is no specific id for an owner, therefore it'll be null. If it ends up being a personal grocery list, there needs to be an owner id to do security checks, especially if we want to implement the idea of personA making a grocery list that is visible to personB but personB cannot add or remove items from a grocery list. Thoughts?

**Pull request merged**